plugins/modules/zos_job_query.py

```diff
@@ -38,6 +38,8 @@
        - A job name can be up to 8 characters long.
        - The I(job_name) can contain include multiple wildcards.
        - The asterisk (`*`) wildcard will match zero or more specified characters.
+       - Note that using this value will query the system for '*' and then return just matching values
+       - This may lead to security issues if there are read-access limitations on some users or jobs
```

Comment:
```suggestion
       - This may lead to security issues if there are read-access limitations on some users or jobs.
```

Reply:
Plus question, what are the security issues? I imagine if a user don't have access to a job it would just not appear in the list returned by ZOAU. This is not a blocker or change request just a question.

Reply:
That was part of the original complaint: user got security warnings because the query was looking at jobs that were restricted from the ansible account.  It turned out when job.py was asking zoau for job_name="something", it was reduced to job_name='*'.  It's rare, but some job classes are not supposed to be viewed by all users (some kind of SAF setting).